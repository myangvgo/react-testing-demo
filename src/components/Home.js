import React from 'react';

export const Home = () => {
    return (
        <div>
            <h1>Home Page</h1>
            <p>This is the home page</p>
            <span>Updated</span>
        </div>
    );
};
