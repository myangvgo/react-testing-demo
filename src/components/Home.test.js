import React from 'react';
// import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Home } from './Home';

// // Basic Test with React-test-renderer
// it('renders correctly react-test-renderer', () => {
//     const renderer = new ShallowRenderer();
//     renderer.render(<Home />);
//     const result = renderer.getRenderOutput();

//     expect(result).toMatchSnapshot();
// });

it('renders correctly enzyme', () => {
    const wrapper = shallow(<Home />);

    expect(toJson(wrapper)).toMatchSnapshot();
});
