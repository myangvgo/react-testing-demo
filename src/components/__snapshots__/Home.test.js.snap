// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<div>
  <h1>
    Home Page
  </h1>
  <p>
    This is the home page
  </p>
  <span>
    Updated
  </span>
</div>
`;
